core/src/main/scala/org/apache/spark/deploy/security/HadoopFSDelegationTokenProvider.scala

```diff
@@ -145,6 +145,12 @@ private[deploy] class HadoopFSDelegationTokenProvider
         interval
       }.toOption
     }
+    // cancel the temporary delegation tokens to avoid leakage
+    creds.getAllTokens.asScala.foreach {
+      case id: AbstractDelegationTokenIdentifier => id.cancel(hadoopConf)
```

Comment:
`cancel` throws `IOException` and `InterruptedException`. Let's say we have 2+ tmp tokens and the following happens:
* First token `cancel` throws an exception
* All the rest is just skipped to be canceled
* Leak partially solved

All in all it would be good to handle cancels independently with catch part + adding some error log into the catch area including the exception content.

Reply:
Agree, can you make this change @jojochuang ?

Reply:
@jojochuang can you make the change to merge this? You've made the majority of the work already :)